### PR TITLE
Encrypted image

### PR DESCRIPTION
**Pull request by stevenhorsman**
PoC test for encrypted image support

### CODE REVIEW

File: tests/integration/kubernetes/k8s-guest-pull-image-encrypted.bats [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    if [ "${KBS}" = "false" ]; then
+        skip "Test skipped as KBS not setup"
+    fi
+
+    if ! is_confidential_hardware; then
+        kbs_set_allow_all_resources
+    fi
+
+    kbs_set_resource "default" "key" "1" "${decryption_key}"
```
Comment (Xynnn007) [inline tests/integration/kubernetes/k8s-guest-pull-image-encrypted.bats:41]:
```suggestion
    kbs_set_resource_base64 "default" "key" "ssh-demo" "${decryption_key}"
```

Original `kbs_set_resource` function does not support that the content of resource is binary, but the mage decryption key is in binary. Thus we need this new function.

File: src/agent/kata-agent.service.in [hunk 2]
```diff
@@ -14,6 +14,7 @@ Wants=kata-containers.target
 # from a VM vsock port
 StandardOutput=tty
 Type=simple
+Environment=OCICRYPT_KEYPROVIDER_CONFIG=/etc/ocicrypt_config.json
```
Comment (Xynnn007) [inline src/agent/kata-agent.service.in:17]: I will change this way to follow the old ways, s.t. kata-agent is responsible for putting `ocicrypt_config.json`.

**Review by Xynnn007 — COMMENTED** [review]
Let me update `ocicrypt_config.json` putting in #9382

**Comment by stevenhorsman** [comment]
Merged into https://github.com/kata-containers/kata-containers/pull/9382, so closing it for now

**Pull request closed without merging**